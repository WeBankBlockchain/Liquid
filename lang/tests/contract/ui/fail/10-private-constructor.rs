use liquid_lang as liquid;

#[liquid::contract]
mod noop {
    #[liquid(storage)]
    struct Noop {}

    #[liquid(methods)]
    impl Noop {
        fn new(&mut self) {}

        pub fn noop(&self) {}
    }
}

fn main() {}
