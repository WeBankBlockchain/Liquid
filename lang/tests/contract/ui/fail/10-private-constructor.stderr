error: the visibility for contract constructor should be `pub`
  --> $DIR/10-private-constructor.rs:10:12
   |
10 |         fn new(&mut self) {}
   |            ^^^
